airflow/www/static/js/calendar.js

```diff
@@ -0,0 +1,254 @@
+      .attr("transform", `translate(0, ${monthHeaderHeight})`)
+      .selectAll("g")
+      .data(d3.range(7))
+      .enter()
+      .append('text')
+      .attr("y", i => (i + 0.5) * cellSize)
+      .attr("dy", "0.31em")
+      .attr("font-size", 12)
+      .text(formatDay);
+
+    // create months groups to old the individual day cells
+    const months = yearBody
+      .append("g")
+      .attr("transform", data => `translate(10, 0)`);
+
+    const startWeekOffset = (y, m) => m == 0 ? 0 : toMoment(y, m, 1).isoWeek()
+    const endWeekOffset = (y, m) => toMoment(y, m, 1).add(1, 'month').subtract(7, 'day').isoWeek() + 1
+    const monthXPos = (y, m) => startWeekOffset(y, m) * cellSize + m * cellSize
+    const monthWitdth = (y, m) => (endWeekOffset(y, m) - startWeekOffset(y, m) + 1) * cellSize
+
+    const month = months
+      .append("g")
+      .selectAll("g")
+      .data(data => d3
+        .range(12)
+        .map(i => {
+          const year = data.year
+          const month = i
+          const dagStatesByDay = data.dagStates[month] || {}
+          return {
+            'year': year,
+            'month': month,
+            'dagStates': dagStatesByDay,
+          }
+        })
+      )
+      .enter()
+      .append('g')
+      .attr("transform", data => `translate(${monthXPos(data.year, data.month)}, 0)`);
+
+    // write month names
+    month
+      .append('text')
+      .attr("x", data => monthWitdth(data.year, data.month) / 2)
+      .attr("y", monthHeaderHeight / 2)
+      .attr("text-anchor", "middle")
+      .text(data => formatMonth(data.month))
+
+    const monthBody =  month
+      .append("g")
+      .attr("transform", data => `translate(0, ${monthHeaderHeight})`)
+
+    // create days inside the month groups
+    const tipHtml = data => {
+      const stateCounts = d3.entries(data.dagStates).map(kv => `${kv.value[0].count} ${kv.key}`)
+      const d = toMoment(data.year, data.month, data.day)
+      return `<strong>${d}</strong><br>${stateCounts.join('<br>')}`
+    }
+
+    monthBody
+      .selectAll("text")
+      .data(data => d3
+        .range(daysInMonth(data.year, data.month))
+        .map(i => {
+          const day = i + 1
+          const dagStatesByState = data.dagStates[day] || {}
+          return {
+            'year': data.year,
+            'month': data.month,
+            'day': day,
+            'dagStates': dagStatesByState,
+            'weekDay': toMoment(data.year, data.month, day).day(),
+            'monthWeek': weekOfMonth(data.year, data.month, day),
+          }
+        })
+      )
+      .enter()
+      .append('rect')
+      .attr("x", data => data.monthWeek  * cellSize)
+      .attr("y", data => data.weekDay  * cellSize)
+      .attr("width", cellSize)
+      .attr("height", cellSize)
+      .attr("class", data => stateClass(data.dagStates))
+      .style({
+        "fill": "#eeeeee",
+        "stroke": "#ffffff",
+        "stroke-width": 1,
+        "cursor": "pointer"
```

Comment:
I would be nice to abstract most of the inline styling in this file into a CSS file.